### PR TITLE
Non-algolia search support

### LINKED ISSUE #169: Non-algolia search support (ootb search with no external deps)
When running my site I want search either using something fancy like algolia (if i qualify and can setup) or using something without dependencies so that people can find things quickly

Follow up to #34 where we implemented algolia.

Suggest we move to using pliny search setup: https://github.com/timlrx/pliny/tree/main/packages/pliny/src/search (see more discussion below)

## Acceptance

- [ ] Algolia still works
- [ ] Have non-algolia "built-in" search (following pliny probably)
- [ ] Updated https://flowershow.app/docs/search 

## Analysis

I suggest we move to using pliny approach to search which supports algolia and inbuilt solution - see https://github.com/timlrx/pliny/issues/26. Actual implementation is in https://github.com/timlrx/pliny/pull/35

As this is very well designed and modularized i believe we can largely reuse the component quite cleanly, see e.g. https://github.com/timlrx/pliny/pull/35/files#diff-3d648d6c4a1017daf334b58014aac23abd21ad0c73e208af53e6640bc6223c80R16 where in `_app.tsx` we have:

```jsx
import { SearchProvider } from 'pliny/search'

...

      <SearchProvider searchConfig={siteMetadata.search}>
        <LayoutWrapper>
          <Component {...pageProps} />
        </LayoutWrapper>
      </SearchProvider>
```

### PR DESCRIPTION
**Pull request by khalilcodes**
### Summary
Tasks from #169 

This PR adds the feature to setup either a kbar or algolia search component using the config file.

### Changes
* Search components setup using pliny approach (context, provider)
* Add search provider in config and delete env files
* Add postbuild script to generate `public/search.json` for local search support
* update docs for local and algolia setup @ [docs/search](https://deploy-preview-248--spectacular-dragon-c1015c.netlify.app/docs/search)

**Keeping Algolia as search provider for flowershow since it's already setup on Netlfiy**
**Kbar requires some debugging on netlify build. More details in issue**

### Screenshot (local search using kbar)
Screen Shot 2022-11-03 at 11 10 13 AM

### CODE REVIEW

**Comment by khalilcodes** [comment]
Hey @olayway do you have any idea why E2E test for `import { allDocuments } from "contentlayer/generated"` in `pages/[[...slug]].jsx`  is failing ?

File: site/content/docs/search.md [hunk 1]
```diff
@@ -2,34 +2,57 @@
...
 
 Flowershow supports search functionality on the website to deliver realtime results for content available on all your pages. Clicking the result will lead the user to the relevant page.
 
 It is available in the navbar or can also be accessed by pressing the Ctrl+K (⌘+K on macOS) keyboard shortcut to open the search modal.
 
-## Developers
+## Adding the search component
+
+You can either add a local search component or by using the external algolia search provider. These can be setup in your `config.js` file.
+
```
Comment (olayway) [inline site/content/docs/search.md:14]: It's not really clear what you mean by "local search component". Can I create some custom component? It's actually either Algolia or kbar. Also, let's maybe add a sentence or two about how are they different? E.g. (if I understand correctly) algolia search allows for full-text search with external indexing and in kbar you don't need an external provider but you can only do keyword search?

File: templates/default/components/search/KBar.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import Router from "next/router";
+
+import { KBarModal } from "./KBarModal";
+
+export function KBarSearchProvider({ kbarConfig, children }) {
+  const { searchDocumentsPath, defaultActions } = kbarConfig;
```
Comment (olayway) [inline templates/default/components/search/KBar.jsx:6]: What are the `defaultActions` for? Should it be mentioned in the docs?

**Comment by olayway** [comment]
@khalilcodes you may also want to add `public/search.json` to `.gitignore`

**Comment by olayway** [comment]
> Hey @olayway do you have any idea why E2E test for `import { allDocuments } from "contentlayer/generated"` in `pages/[[...slug]].jsx` is failing ?

@khalilcodes I'm not sure why this error looked like this because locally it said sth different, but there was one issue related to inconsistency between named and default exports in the E2E fixture config.js. I've fixed it on the main branch. Let's see if it works it you rebase. I've tested it locally on your branch and it should work.

**Comment by khalilcodes** [comment]
@olayway not sure if I did the rebase correctly and the commits have repeated here I think ?

**Comment by olayway** [comment]
> @olayway not sure if I did the rebase correctly and the commits have repeated here I think?

@khalilcodes rebase looks good to me, all your commits are now after the monorepo PR merge to main. I haven't fixed the github workflows yet so don't worry about the failing tests. If you have made some adjustments and this is good to go you can re-request my review, I'll quickly check it and we'll merge this.

**Comment by khalilcodes** [comment]
@olayway using algolia now. Also please check [docs/search](https://deploy-preview-248--spectacular-dragon-c1015c.netlify.app/docs/search) for instructions on `defaultActions` in kbar if it looks okay to understand.

**Comment by olayway** [comment]
> @olayway using algolia now. Also please check [docs/search](https://deploy-preview-248--spectacular-dragon-c1015c.netlify.app/docs/search) for instructions on `defaultActions` in kbar if it looks okay to understand.

@khalilcodes the docs page is great! The only thing I think we could add is where do the search keywords come from - if I remember correctly they are taken from the page description in the frontmatter?

**Comment by olayway** [comment]
@khalilcodes hmm sth is off with the files being shown as changed now. Previously there were only the files you were working on and now github shows there are ~170 files changed 😅

**Review by olayway — APPROVED** [review]

**Comment by olayway** [comment]
@khalilcodes yes, let's merge it and create a separate issue for kbar

**Comment by olayway** [comment]
@khalilcodes btw, don't worry about the failing tests, the workflows are fixed on the main branch so they should pass when we merge it.

**Pull request merged**